File: FRONTEND_BuildingRegistration/BuildingRegistration/src/app/_models/Building.ts
import { FloorDTO } from "./Floor"
import { GarageDTO } from "./Garage"
import { GarageSectionDTO } from "./GarageSection"
import { Option } from "./options"

export interface BuildingResponse{
    buildingName?:string
    buildingAddress?:string

    buildingHasGuard?:string
    buildingHasCleaningService?:string
    buildingHasJanitor?:string
    buildingHasWifi?:string
    buildingAllowPet?:string

    buildingElevatorQty?:number
    buildingNote?:string
    
    floorDtos?:FloorDTO[],
    garageDtos?:GarageDTO[]
}

export interface Building{
    buildingName?:string
    buildingAddress?:string

    buildingHasGuard?:boolean
    buildingHasCleaningService?:boolean
    buildingHasJanitor?:boolean
    buildingHasWifi?:boolean
    buildingAllowPet?:boolean

    buildingElevatorQty?:number
    buildingNote?:string
}

export interface BuildingDTO extends Building{
    floorDtos?:FloorDTO[],
    garageDtos?:GarageDTO[]
}

export interface BuildingForOccupant{
    buildingName: string,
    buildingAddress: string,
    floorName: string,
    departmentName: string,
    garageLevel: string,
    garageSections:GarageSectionDTO[]
}

File: BACKEND_BuildingRegistration/BuildingRegistrationAPI_DL/Models/Building.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BuildingRegistrationAPI_DL.Models
{
    public class Building
    {
        public int BuildingId { get; set; }
        public string? BuildingName { get; set; }
        public string BuildingAddress { get; set; }
        public bool? BuildingHasGuard { get; set; }
        public bool? BuildingHasCleaningService { get; set; }
        public bool? BuildingHasJanitor { get; set; }
        public bool? BuildingHasWifi { get; set; }
        public bool? BuildingAllowPet { get; set; }
        public short BuildingElevatorQty { get; set; } 
        public string? BuildingNote { get; set; }
    }
}
